src/components/scenes/RequestScene.js

```diff
@@ -225,8 +226,15 @@ export class Request extends Component<Props, State> {
     }
 
     const color = 'white'
-    const { primaryCurrencyInfo, secondaryCurrencyInfo, exchangeSecondaryToPrimaryRatio, onSelectWallet } = this.props
+    const { primaryCurrencyInfo, secondaryCurrencyInfo, exchangeSecondaryToPrimaryRatio, onSelectWallet, wallets } = this.props
     const requestAddress = this.props.useLegacyAddress ? this.state.legacyAddress : this.state.publicAddress
+    const walletsCopy = { ...wallets }
```

Comment:
Logic seems a bit backwards. Instead of copying everything then deleting the inactive wallets. Why not just iterate over all wallets and copy the ones that are activated.